st2common/tests/unit/services/test_rbac.py

```diff
@@ -203,6 +205,47 @@ def test_grant_and_revoke_role(self):
         role_dbs = user_db.get_roles()
         self.assertItemsEqual(role_dbs, [])
 
+    def test_grant_duplicate_role(self):
+        user_db = UserDB(name='test-user-1')
+        user_db = User.add_or_update(user_db)
+
+        # Initial state, no roles
+        role_dbs = rbac_services.get_roles_for_user(user_db=user_db)
+        self.assertItemsEqual(role_dbs, [])
+
+        role_dbs = user_db.get_roles()
+        self.assertItemsEqual(role_dbs, [])
+
+        # Assign a role, should have one role assigned
+        rbac_services.assign_role_to_user(
+            role_db=self.roles['custom_role_1'], user_db=user_db,
+            source='assignments/%s_1.yaml' % user_db.name)
+
+        role_dbs = rbac_services.get_roles_for_user(user_db=user_db)
+        self.assertItemsEqual(role_dbs, [self.roles['custom_role_1']])
+
+        role_dbs = user_db.get_roles()
+        self.assertItemsEqual(role_dbs, [self.roles['custom_role_1']])
+
+        # Assign the same role again.
+        rbac_services.assign_role_to_user(
+            role_db=self.roles['custom_role_1'], user_db=user_db,
+            source='assignments/%s_2.yaml' % user_db.name)
+
+        role_dbs_2 = rbac_services.get_roles_for_user(user_db=user_db)
+        self.assertItemsEqual(role_dbs_2, [self.roles['custom_role_1']])
+
+        role_dbs = user_db.get_roles()
+        self.assertItemsEqual(role_dbs_2, [self.roles['custom_role_1']])
+
+        # Revoke previously assigned role, should have no roles again
+        rbac_services.revoke_role_from_user(role_db=self.roles['custom_role_1'], user_db=user_db)
```

Comment:
I'm a bit confused about this one - just wondering how much sense `revoke_role_from_user` even makes now.

In this case it makes sense to remove all the assignments for a particular user, but not sure about the syncer case. If user is assigned role X from two different assignments and one of those assignments files is deleted, this role should still be there, right? I thought that's that part we are solving.